intersections/build.boot

```diff
@@ -0,0 +1,113 @@
+(set-env!
+ :repositories #(conj % '["bintray" {:url "http://dl.bintray.com/nitram509/jbrotli"}])
+ :resource-paths #{"resources"}
+ :dependencies '[[cljsjs/boot-cljsjs "0.6.0" :scope "test"]
+                 [nha/boot-uglify "0.0.7" :scope "test"]])
+
+(def +version+ "1.0.0-0")
+
+(task-options!
+ pom  {:project     'cljsjs/intersections
+       :version     +version+
+       :description "Intersection is a JavaScript object used to capture the results of the intersection of two Shape objects."
+       :url         "http://www.quazistax.com/testIntersection.html"
+       :scm         {:url "git://github.com/thelonious/kld-intersections.git"}
+       :license     {"BSD" "http://opensource.org/licenses/BSD-3-Clause"}})
+
+(require '[cljsjs.boot-cljsjs.packaging :refer :all]
+         '[nha.boot-uglify.minify-js :as minify-js]
+         '[boot.core :as c]
+         '[boot.util :as u]
+         '[boot.file :as file]
+         '[clojure.java.io :as io])
+
+(defn normalize-path
+  "Adds a / if missing at the end of the path."
+  [path]
+  (if-not (= \/ (last path))
+    (str path "/")
+    path))
+
+(c/deftask uglify
+  "Create the minified file
+
+  This task also copies the minified file in the development folder."
+  [_ working-dir DIR str "Path where the js files to uglify will be in"
+   _ file-name NAME [str] "Vector of JS source file names to uglify (order counts)"
+   _ prod-dir DIR str "The package production dir"
+   _ dev-dir DIR str "The package development dir"
+   _ prod-name NAME str "File name for the minified file"
+   _ dev-name NAME str "File name for the development file"]
+  (let [tmp (c/tmp-dir!)]
+    (c/with-pre-wrap fileset
+      (let [real-dir (->> (c/ls fileset)
+                          (c/by-re [(re-pattern working-dir)])
+                          first
+                          c/tmp-file
+                          .getParent
+                          normalize-path)
+            prod-file (io/file tmp (str prod-dir prod-name))
+            dev-file (io/file tmp (str dev-dir dev-name))]
+        (u/dbug* "Minifying content of: %s\n" real-dir)
+        (let [uglify-result (minify-js/minify-js (mapv #(str real-dir %) file-name)
+                                                 prod-file)]
+          (u/dbug* "Uglify result: %s\n" (u/pp-str uglify-result)))
+        (u/dbug* "Minified file: %s\n" prod-file)
+        (u/dbug* "Development file: %s\n" dev-file)
+        (io/make-parents dev-file)
+        (file/copy-atomically prod-file dev-file)
+        (-> fileset (c/add-resource tmp) c/commit!)))))
+
+(deftask package []
+  (let [uglify-dir (normalize-path "uglify/")
+        prod-dir (normalize-path "cljsjs/intersections/production/")
+        dev-dir (normalize-path "cljsjs/intersections/development/")]
+    (comp
+     (download :url "https://raw.githubusercontent.com/thelonious/kld-intersections/development/lib/Intersection.js"
```

Comment:
Please use versioned url to download the archive or the files. Use the `+lib-version+` var you have defined to build the url:

```clj
(def +lib-version+ "1.2.3")

...

(download :url (format "https://github.com/example/project/archive/%s.zip" +lib-version+)
          :unzip true)
```

If the library for some reason doesn't use git tags and doesn't have versioned downloads, you can use git sha to download a specific version. Use git sha for the commit which released the version.

```clj
(def +lib-version+ "1.2.3")
(def +git-sha+ "97ffda1f0e1d45fde4dcfbfcef0fcd904c714918")

...

(download :url (format "https://github.com/example/project/archive/%s.zip" +git-sha+)
          :unzip true)
```

Using versioned downloads ensures that the builds are reproductible and that the CI version will use the correct archive/files to build and deploy the package.